templates/shared/repo_search.tmpl

```diff
@@ -0,0 +1,75 @@
+<div class="ui secondary filter menu gt-ac gt-mx-0">
+	<form class="ui form ignore-dirty gt-f1">
+		<input type="hidden" name="sort" value="{{$.SortType}}">
+		<input type="hidden" name="language" value="{{$.Language}}">
+		<div class="ui fluid action input">
+			{{template "shared/searchinput" dict "Value" .Keyword}}
+			{{if .PageIsExploreRepositories}}
+				<input type="hidden" name="only_show_relevant" value="{{.OnlyShowRelevant}}">
+			{{else if .TabName}}
+				<input type="hidden" name="tab" value="{{.TabName}}">
+			{{end}}
+			<button class="ui primary button">{{ctx.Locale.Tr "explore.search"}}</button>
+		</div>
+	</form>
+	{{$tabQuery := ""}}
+	{{if .TabName}}{{$tabQuery = printf "tab=%s&" (QueryEscape .TabName)}}{{end}}
+	{{$languageQuery := ""}}
+	{{if .TabName}}{{$languageQuery = printf "language=%s&" (QueryEscape .Language)}}{{end}}
+	{{$queryParams := printf "%s%sq=%s" $tabQuery $languageQuery (QueryEscape .Keyword)}}
+	<!-- Filter -->
+	{{$queryParamsWithSort := printf "%s&sort=%s" $queryParams (QueryEscape .SortType)}}
+	<form class="ui form ignore-dirty" id="repo-search-form" data-query-params="{{$queryParamsWithSort}}">
```

Comment:
After reading the code again, I think it could be simplified to this:

```
<form ...>
{{if .TabName}} < input hidden tab >{{end}}
...
</form>
```

Then no need to construct the "queryParamsWithSort" manually. Just use the JS code to submit this "GET" form.

Reply:
You could try the GET form by this sample:


```
<form action="/link" method="get">
    <input type=hidden name=key value=val>
<form>
```

If you submit the form, the URL is "/link?key=val", the browser does everything for you.